Simperium/SPJSONStorage.m

```diff
@@ -210,22 +208,16 @@ - (void)deleteObject:(id)dict
 }
 
 - (void)deleteAllObjectsForBucketName:(NSString *)bucketName {
-//    NSFetchRequest *fetchRequest = [[NSFetchRequest alloc] init];
-//    NSEntityDescription *entity = [NSEntityDescription entityForName:entityName inManagedObjectContext:context];
-//    [fetchRequest setEntity:entity];
-//    
-//    // No need to fault everything
-//    [fetchRequest setIncludesPropertyValues:NO];
-//    
-//    NSError *error;
-//    NSArray *items = [context executeFetchRequest:fetchRequest error:&error];
-//    
-//    for (NSManagedObject *managedObject in items) {
-//        [context deleteObject:managedObject];
-//    }
-//    if (![context save:&error]) {
-//        NSLog(@"Simperium error deleting %@ - error:%@",entityName,error);
-//    }
+    dispatch_sync(self.storageQueue, ^{
+        // Nuke Bucket Entities from the `allObjects` collection
+        NSDictionary<NSString *, SPObject *> *bucket = [self.objects objectForKey:bucketName];
+        for (NSString *key in bucket.allKeys) {
+            [self.allObjects removeObjectForKey:key];
+        }
+
+        // And now nuke the entire bucket
+        [self.objects removeObjectForKey:bucketName];
+    });
```

Comment:
Can we just recreate `self.objects` and `self.allObjects` instead of removing one by one? Is there something else left in those dictionaries?

Reply:
The API's name for this one is `deleteAllObjectsForBucketName`, which is invoked by `SPBucket.deleteAllObjects`, so we should only delete the objects associated for the specified Bucket Name.

On logout Simperium invokes `deleteAllObjects` on every bucket (and that leads to this invocation)

Reply:
I see, thank you for the explanation!